apps/web/components/connections/ConnectionCard/EditingDisplayName.tsx

```diff
@@ -0,0 +1,120 @@
+import {useMutation, useQueryClient} from '@tanstack/react-query'
+import type {Id} from '@usevenice/cdk-core'
+import {CircularProgress} from '@usevenice/ui'
+import {CloseIcon} from '@usevenice/ui/icons'
+import clsx from 'clsx'
+import {useEffect, useRef, useState} from 'react'
+import {browserSupabase} from '../../../contexts/common-contexts'
+import {useOnClickOutside} from '../../../hooks/useOnClickOutside'
+
+interface EditingDisplayNameProps {
+  displayName: string
+  onCancel: () => void
+  onUpdateSuccess: () => void
+  resourceId: Id['reso']
+}
+
+export function EditingDisplayName(props: EditingDisplayNameProps) {
+  const {onCancel, onUpdateSuccess, resourceId} = props
+  const [displayName, setDisplayName] = useState(props.displayName)
+
+  const queryClient = useQueryClient()
+  const {mutate: updateDisplayName, isLoading: isUpdating} = useMutation(
+    async () => {
+      const {error} = await browserSupabase
+        .from('resource')
+        .update({display_name: displayName})
+        .eq('id', resourceId)
+
+      if (error) {
+        throw new Error(error.message)
+      }
+      return undefined
+    },
+    {
+      mutationKey: ['resource', 'update', resourceId],
+      onSuccess: async () => {
+        await queryClient.invalidateQueries({
+          queryKey: ['connections', 'list'],
+        })
```

Comment:
@tonyxiao I was thinking; in the future iteration, it'd probably feel almost instantaneous if we get the updated response back (doing [`.select()`](https://supabase.com/docs/reference/javascript/update)) and use that to [update the cache](https://tanstack.com/query/v4/docs/react/guides/updates-from-mutation-responses).